src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleConnectionParameters.java

```diff
@@ -114,6 +136,40 @@ private String generateUrl() {
       buf.append("/");
       buf.append(serviceName);
     }
+
     return buf.toString();
   }
+
+  private String generateSecureUrl(StringBuilder buf) {
+    buf.append("(DESCRIPTION=");
```

Comment:
The StringBuffer should be refactored in an Oracle TNS Entry builder class.

Reply:
Added builder class, very basic. Eventually the user will be allowed to add properties to a map and the builder will work with it